independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/InstanceHandleImpl.java

```diff
@@ -61,10 +60,14 @@ public T get() {
     @Override
     public void destroy() {
         if (instance != null && destroyed.compareAndSet(false, true)) {
-            if (bean.getScope().equals(Dependent.class)) {
-                destroyInternal();
+            if (destroyLogic != null) {
+                destroyLogic.accept(instance);
             } else {
-                Arc.container().getActiveContext(bean.getScope()).destroy(bean);
+                if (bean.getScope().equals(Dependent.class)) {
+                    destroyInternal();
+                } else {
+                    Arc.container().getActiveContext(bean.getScope()).destroy(bean);
```

Comment:
`getActiveContext` can return `null` if no context is active, we should probably check that.

Reply:
Good point. I did not check the existing code...